ruby/red-arrow/lib/arrow/slicer.rb

```diff
@@ -351,5 +356,22 @@ def evaluate
         BooleanArray.new(raw_array)
       end
     end
+
+    class MatchSubstringFamilyCondition < Condition
+      def initialize(function, column, substring, ignore_case)
+        @function = function
+        @column = column
+        @options = MatchSubstringOptions.new
+        @options.pattern = substring
+        @options.ignore_case = ignore_case
+      end
+
+      def evaluate
+        case @function
+        when "match_substring"
+          Function.find("match_substring").execute([@column.data], @options).value
+        end
```

Comment:
```suggestion
        Function.find(@function).execute([@column.data], @options).value
```

Reply:
Thanks, I will do so. (May some change needed to implement `#!@` or other family methods.)